platform/darwin/src/MGLImageSource.h

```diff
@@ -0,0 +1,95 @@
+#import "MGLSource.h"
+
+#import "MGLFoundation.h"
+#import "MGLTypes.h"
+#import "MGLGeometry.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+MGL_EXPORT
+/**
+ `MGLImageSource` is a content source that is used for a georeferenced raster
+ image to be shown on the map. The georeferenced image scales and rotates as the
+ user zooms and rotates the map. Images may also be used as icons or patterns 
+ in a style layer. To register an image for use as an icon or pattern, 
+ use the `-[MGLStyle setImage:forName:]` method. To configure a point 
+ annotation’s image, use the `MGLAnnotationImage` class.
+ 
+ The geographic location of the raster image content, supplied with 
+ `MGLCoordinateQuad`, can be non-axis aligned.
+ `MGLImageSource` supports raster content from `NSURL`, `NSImage` (macOS), or 
+ `UIImage` (iOS).
+ An image source is added to an `MGLStyle` object along with one or more
+ `MGLRasterStyleLayer` objects. Use a raster style layer to control the
+ appearance of content supplied by the image source.
+ 
+ Each
+ <a href="https://www.mapbox.com/mapbox-gl-style-spec/#sources-image"><code>image</code></a>
+ source defined by the style JSON file is represented at runtime by an
+ `MGLImageSource` object that you can use to initialize new style layers. You
+ can also add and remove sources dynamically using methods such as
+ `-[MGLStyle addSource:]` and `-[MGLStyle sourceWithIdentifier:]`.
+
+ ### Example
+
+ ```swift
+ let coordinates: MGLCoordinateQuad = MGLCoordinateQuad(
+   topLeft: CLLocationCoordinate2D(latitude: 46.437, longitude: -80.425),
+   bottomLeft: CLLocationCoordinate2D(latitude: 37.936, longitude: -80.425),
+   bottomRight: CLLocationCoordinate2D(latitude: 37.936, longitude: -71.516),
+   topRight: CLLocationCoordinate2D(latitude: 46.437, longitude: -71.516))
+ let source = MGLImageSource(identifier: "radar", coordinateQuad: coordinates, imageURL: URL(string: "https://www.mapbox.com/mapbox-gl-js/assets/radar.gif")!)
+ mapView.style?.addSource(source)
+ ```
+ */
+MGL_EXPORT
+@interface MGLImageSource : MGLSource
+
+#pragma mark Initializing a Source
+
+/**
+ Returns a georeferenced image source with an identifier, coordinates and a URL.
+
+ @param identifier A string that uniquely identifies the source.
+ @param coordinateQuad the top left, top right, bottom right, and bottom left coordinates for the image.
+ @param url An HTTP(S) URL, absolute file URL, or local file URL relative to the
+    current application’s resource bundle.
+ @return An initialized shape source.
+ */
+- (instancetype)initWithIdentifier:(NSString *)identifier coordinateQuad:(MGLCoordinateQuad)coordinateQuad imageURL:(NSURL *)url;
```

Comment:
This will bridge to Swift as `init(identifier:coordinateQuad:image:)`, which is a bit misleading. (Since Objective-C conventionally describes the type in argument labels where Swift omits them, the Swift compiler tries to remove such words from argument labels when bridging Objective-C methods.) We can prevent this situation by replacing `imageURL:` with `URL:`, which is also consistent with `-[MGLShapeSource initWithIdentifier:URL:options:]`.